tripal_chado/src/Plugin/Field/FieldType/ChadoContactByRoleTypeDefault.php

```diff
@@ -117,7 +117,7 @@ public static function tripalTypes($field_definition) {
     $linker_pkey_col = $linker_schema_def['primary key'];
     $terms['linker_pkey'] = $terms['record_id'];
     //    - left table foreign key
-    $linker_left_col = array_keys($linker_schema_def['foreign keys'][$base_table]['columns'])[0];
+    $linker_left_col = self::getChadoForeignKeyColumn($linker_table, $base_table, $schema);
```

Comment:
This field is weird and uses `$schemaObj` instead of `$schema` 🙈 

```suggestion
    $linker_left_col = self::getChadoForeignKeyColumn($linker_table, $base_table, $schemaObj);
```